bin/helpers/usageReporting.js

```diff
@@ -0,0 +1,209 @@
+    if (!version) {
+      version = get_version('cypress');
+    }
+    return version;
+  } else {
+    return get_version('cypress');
+  }
+}
+
+function bstack_json_found_in_pwd() {
+  try {
+    if (fs.existsSync(path.join(process.cwd(), 'browserstack.json'))) {
+      //file exists
+      return true;
+    }
+    else {
+      return false;
+    }
+  } catch (err) {
+    return null;
+  }
+}
+
+function cypress_json_found_in_pwd() {
+  try {
+    if (fs.existsSync(path.join(process.cwd(), 'cypress.json'))) {
+      //file exists
+      return true;
+    }
+    else {
+      return false;
+    }
+  } catch (err) {
+    return null;
+  }
+}
+
+function npm_global_path() {
+  return cp.execSync('npm root -g', { stdio: 'pipe' }).toString().trim();
+}
+
+function cli_version_and_path(bsConfig) {
+  // 1. check version of Cypress installed in local project
+  // 2. check version of Cypress installed globally if not present in project
+
+  if (bsConfig) {
+    let _path = path.join(bsConfig.run_settings.cypress_proj_dir, 'node_modules', 'browserstack-cypress');
+    let version = get_version(_path);
+    if (!version) {
+      version = get_version('browserstack-cypress');
+
+      if (!version) {
+        // return path = null if version is null
+        return [null, null];
+      }
+      _path = npm_global_path();
+      return [version, _path];
+    }
+    return [version, _path];
+  } else {
+    let version = get_version('browserstack-cypress');
+
+    if (!version) {
+      // return path = null if version is null
+      return [null, null];
+    }
+    return [version, npm_global_path()];
+  }
+}
+
+function ci_environment() {
+  var env = process.env;
+  // Jenkins
+  if ((typeof env.JENKINS_URL === "string" && env.JENKINS_URL.length > 0) || (typeof env.JENKINS_HOME === "string" && env.JENKINS_HOME.length > 0)) {
+    return "Jenkins";
+  }
+  // CircleCI
+  if (env.CI === "true" && env.CIRCLECI === "true") {
+    return "CircleCI"; 
+  }
+  // Travis CI
+  if (env.CI === "true" && env.TRAVIS === "true") {
+    return "Travis CI";
+  }
+  // Codeship
+  if (env.CI === "true" && env.CI_NAME === "codeship") {
+    return "Codeship";
+  }
+  // Bitbucket
+  if (env.BITBUCKET_BRANCH && env.BITBUCKET_COMMIT) {
+    return "Bitbucket";
+  }
+  // Drone
+  if (env.CI === "true" && env.DRONE === "true") {
+    return "Drone";
+  }
+  // Semaphore
+  if (env.CI === "true" && env.SEMAPHORE === "true") {
+    return "Semaphore";
+  }
+  // GitLab
+  if (env.CI === "true" && env.GITLAB_CI === "true") {
+    return "GitLab";
+  }
+  // Buildkite
+  if (env.CI === "true" && env.BUILDKITE === "true") {
+    return "Buildkite";
+  }
+  // Visual Studio Team Services
+  if (env.TF_BUILD === "True") {
+    return "Visual Studio Team Services";
+  }
+  // if no matches, return null
+  return null;
+}
+
+function isUsageReportingEnabled() {
+  return process.env.DISABLE_USAGE_REPORTING;
```

Comment:
Let's document how to opt-out of the usage reporting, please.

Also, it's worth mentioning what all will be collected, and everything is anonymous and cannot be used for identifying a run.